src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -1252,12 +1252,6 @@ public void visitMethodDef(JCMethodDecl tree) {
                         }
                     }
                 }
-                if (m.isConstructor() && m.type.getParameterTypes().size() == 0) {
-                    if ((owner.type == syms.objectType) ||
-                            (tree.body.stats.size() == 1 && TreeInfo.getConstructorInvocationName(tree.body.stats, names, false) == names._super)) {
-                        m.flags_field |= EMPTYNOARGCONSTR;
-                    }
-                }
 
```

Comment:
This is too late here. Now this is moved up to MemberEnter

Reply:
This was in being set in Attribution phase - and is also consulted in the attribution phase. 
so when we have classes like:
```
value class V extends A {}
abstract class A { A() {} } 
```
when we are attributing V, A may not be attributed. While attributing V, we will check if V's superclass (A) meets all the requirements of an abstract class to be a value super but A isn't attributed yet.

What we have done now is to move these analysis to MemberEnter - MemberEnter for V and A are both guaranteed to have finished before Attribution of either starts, so we have eliminated the phase ordering problem